llvm/lib/Target/X86/GISel/X86RegisterBankInfo.cpp

```diff
@@ -68,6 +71,143 @@ X86RegisterBankInfo::getRegBankFromRegClass(const TargetRegisterClass &RC,
   llvm_unreachable("Unsupported register kind yet.");
 }
 
+// Returns whether opcode \p Opc is a pre-isel generic floating-point opcode,
+// having only floating-point operands.
+static bool isPreISelGenericFloatingPointOpcode(unsigned Opc) {
```

Comment:
This should go somewhere generic if we don't have it already (I thought I saw the same switch elsewhere before)

Reply:
We already have something similar in AArch64RegisterBankInfo.cpp / MipsRegisterBankInfo.cpp / PPCRegisterBankInfo.cpp / RISCVRegisterBankInfo.cpp - merging them would be great

Reply:
Is it fine to leave the merging as a follow-up PR? I'll try to send it within this week.

(I just want to keep this separate in case I break some existing code in those targets and people want to revert).

Reply:
Too much worrying, just do it here

Reply:
Apologies it me some time to get back to this. Updated.